### DISCUSSION TITLE
[NB_MODULE]Bindind same module in different source file.

### DISCUSSION THREAD

**Discussion by huangweiwu** [wjakob/nanobind#285] [Q&A] [answered]
Is there any possible to bind one same module in more than one source file?
In other words, the code below appears in tow source file.
```
NB_MODULE(my_module, m)
{
}

**Comment by zhqrbitee** [comment, marked as answer]
Not sure I understand you problem.  
But basically you can do something e.g. below:
```c++
void init_a(nb::module_ &m);

NB_MODULE(my_module, m)
{
    init_a(m);
}
```

```c++
void init_a(nb::module_ &m) {
   // some implementation.
}

```

**Reply by wjakob** [reply]
Please also see the FAQ for details on this.
